packages/moonstone/Scrollable/Scrollable.js

```diff
@@ -687,12 +687,24 @@ class ScrollableBase extends Component {
 
 	onVoice = (e) => {
 		const
-			scroll = e && e.detail && e.detail.scroll,
+			type = e && e.detail && e.detail.scroll,
 			isRtl = this.uiRef.state.rtl,
 			{scrollTop, scrollLeft} = this.uiRef,
 			{maxLeft, maxTop} = this.uiRef.getScrollBounds(),
 			verticalDirection = ['up', 'down', 'top', 'bottom'],
-			horizontalDirection = ['left', 'right', 'leftmost', 'rightmost'];
+			horizontalDirection = ['left', 'right', 'leftmost', 'rightmost'],
+			movement = ['previous', 'next', 'first', 'last'];
+
+		let scroll, index;
+		if (verticalDirection.includes(type) || horizontalDirection.includes(type)) {
+			scroll = type;
+		} else if ((index = movement.indexOf(type)) > -1) {
+			if (this.props.direction === 'horizontal') {
+				scroll = isRtl ? horizontalDirection[index % 2 ? index - 1 : index + 1] : horizontalDirection[index];
+			} else {
+				scroll = verticalDirection[index];
+			}
+		}
```

Comment:
Hi @changgilee 
Is it more clear to replace the code above with the following code? Please let me know your thought?
```
		const
			isRtl = this.uiRef.state.rtl,
			isHorizontal = isRtl && this.props.direction === 'horizontal',
			{scrollTop, scrollLeft} = this.uiRef,
			{maxLeft, maxTop} = this.uiRef.getScrollBounds(),
			verticalDirection = ['up', 'down', 'top', 'bottom'],
			horizontalDirection = isRtl  ? ['right', 'left', 'rightmost', 'leftmost'] : ['left', 'right', 'leftmost', 'rightmost'],
			movement = ['previous', 'next', 'first', 'last'];
		let
			scroll = e && e.detail && e.detail.scroll,
			index = movement.indexOf(scroll));

		if (index > -1) {
			scroll = isHorizontal ? horizontalDirection[index] : verticalDirection[index];
		}
```

Reply:
It seems more clear and simple way. I did update base on your code. One different thing is **isRtl** seem don't need to **isHorizontal** variable because **horizontalDirection** already divided base on **isRtl**